File: app/src/main/java/com/skyline/app/PromotionAdapter.java
package com.skyline.app;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.skyline.app.databinding.ItemPromotionBinding;
import com.skyline.model.Promotion;
import java.util.List;

public class PromotionAdapter extends RecyclerView.Adapter<PromotionAdapter.PromotionViewHolder> {
    private List<Promotion> items;
    private OnItemClickListener listener;

    public interface OnItemClickListener {
        void onItemClick(Promotion item);
    }

    public PromotionAdapter(List<Promotion> items, OnItemClickListener listener) {
        this.items = items;
        this.listener = listener;
    }

    @NonNull
    @Override
    public PromotionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemPromotionBinding binding = ItemPromotionBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new PromotionViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull PromotionViewHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class PromotionViewHolder extends RecyclerView.ViewHolder {
        private ItemPromotionBinding binding;

        public PromotionViewHolder(ItemPromotionBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(Promotion item) {
            String url = item.getImageUrl();
            if (url != null && !url.isEmpty()) {
                Glide.with(binding.imgPromo.getContext())
                        .load(url)
                        .placeholder(R.drawable.img_brand_banner)
                        .error(R.drawable.img_brand_banner)
                        .into(binding.imgPromo);
            } else if (item.getImageRes() != 0) {
                binding.imgPromo.setImageResource(item.getImageRes());
            } else {
                binding.imgPromo.setImageResource(R.drawable.img_brand_banner);
            }

            binding.tvPromoTitle.setText(item.getTitle());
            binding.tvPromoDate.setText(item.getDate());
            binding.getRoot().setOnClickListener(v -> listener.onItemClick(item));
        }
    }
}

File: app/src/main/java/com/skyline/app/PromotionAdapter.kt
package com.skyline.app

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.skyline.app.databinding.ItemPromotionBinding
import com.skyline.model.Promotion

class PromotionAdapter(
    private val items: List<Promotion>,
    private val onClick: (Promotion) -> Unit = {}
) : RecyclerView.Adapter<PromotionAdapter.PromotionViewHolder>() {

    inner class PromotionViewHolder(
        private val binding: ItemPromotionBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(item: Promotion) = with(binding) {
            imgPromo.setImageResource(item.imageRes)
            tvPromoTitle.text = item.title
            tvPromoDate.text = item.date
            root.setOnClickListener { onClick(item) }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PromotionViewHolder {
        val binding = ItemPromotionBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return PromotionViewHolder(binding)
    }

    override fun onBindViewHolder(holder: PromotionViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int = items.size
}